### PR TITLE
Allow to disable bug reference carry over on job group level

### PR DESCRIPTION
**Pull request by Martchus**
* The default (carry over enabled) doesn't change of course
* See https://progress.opensuse.org/issues/48182
* Since I was already touching the code, I've also added 2 other small improvements (see further commits).

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm [hunk 1]
```diff
@@ -96,11 +96,14 @@ sub find_group {
     return $group;
 }
 
+has _cached_properties => undef;
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: Private attribute accessors are not very useful, using `$self->{cached_properties}` is much more efficient. But if you really want to use one it should be declared at the top of the file.

Comment (Martchus) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: Changed to `$self->{cached_properties}`.

Comment (richiejp) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: So it seems you are no longer using the accessor, but forgot to delete the definition?

Comment (Martchus) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: Yes. I've removed it now.

**Comment by Martchus** [comment]
I have fixed the mentioned issues. So I would merge if the CI check completes. (Previously the full-stack test failed but that was likely unrelated.)

**Pull request merged**